### PR TITLE
Feature/homepage

### PR DESCRIPTION
**Pull request by amorea9**
DONE - note that the routing for createEnsemble and createPost is not done yet for now as I'd need to know where we're getting there from to define the route and if any params need to be brought over.

21/11:
- finished footer w/ links and routing
- finished modal popUp w/link and routing
- added facebook button too but we can remove it if we don't need it
- added page for login/register. based on which button the user clicks, components can be rendered conditionally.
- tweaked style all around for the page

*previous progress*
WIP - features to be finished
14/11:
- created routing for almost all pages
- created navigation bar component

### CODE REVIEW

**Review by andreaDiClaudio — APPROVED** [review]

File: client/src/components/molecules/Hero.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            icon={ICON_NAMES.author_icon}
+            iconHeight={13.887}
+            iconWidth={13.887}
+            iconViewbox={"0 0 13.887 13.887"}
+            onClick={() => !loginStatus && displayPopUp(true)}
+            to={loginStatus ? "/posts" : undefined}
+            className="no-underline w-full"
+          ></Button>
+          <Button
+            buttonState="default"
```
Comment (nikidkny) [inline client/src/components/molecules/Hero.tsx:41]: Not a big thing and maybe not even for this pr but I think we will need to remove the state from the button to be a mandatory prop as it is not needed only in special cases I guess

File: client/src/components/molecules/Hero.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        {/* {popUp && <div className="fixed inset-0 bg-black bg-opacity-50 z-30"></div>} */}
+
+        {popUp && (
+          <div className="fixed inset-0 z-50 flex items-center justify-center bg-black/50" onClick={() => displayPopUp(false)}>
+            <div className="bg-white p-6 rounded-lg shadow-lg w-11/12 max-w-md relative mx-2 flex flex-col items-center gap-6">
+              {/* Modal content */}
+              <div className="border-0 border-b border-solid border-gray-300 pb-3">
+                <TextBody size="lg" variant="strong" className="text-blue-500">
+                  {" "}
```
Comment (nikidkny) [inline client/src/components/molecules/Hero.tsx:63]: I don't think you need that space thing here

File: client/src/components/molecules/Hero.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                  Log in to find musicians you can play with in all of Denmark
+                </TextBody>
+              </div>
+              <div className="flex flex-col items-center gap-3 w-full items-stretch">
+                <a href="https://facebook.com" target="_blank">
```
Comment (nikidkny) [inline client/src/components/molecules/Hero.tsx:68]: can we not have the link on the actual button component?

**Review by nikidkny — APPROVED** [review]
Looks good to me, I made small comments but those are nothing blocking. One thing I can see as a bigger thing is that there is a difference in code formatting somehow maybe we should check why is that

File: client/src/components/molecules/Hero.tsx (hunk 1 above)
Comment (amorea9) [reply client/src/components/molecules/Hero.tsx:41]: Agree with this, we can add it in the next branch/feature

File: client/src/components/molecules/Hero.tsx (hunk 3 above)
Comment (amorea9) [reply client/src/components/molecules/Hero.tsx:68]: Let me try something really quick :P

**Review by andreaDiClaudio — APPROVED** [review]

**Pull request merged**